pkg/machines/hostvmslist.jsx

```diff
@@ -463,36 +467,78 @@ Vm.propTypes = {
 /**
  * List of all VMs defined on this host
  */
-const HostVmsList = ({ vms, config, dispatch }) => {
-    if (vms.length === 0) {
+class HostVmsList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.deviceProxyHandler = this.deviceProxyHandler.bind(this);
+        this.state = { hostDevices: []};
+        this.client = cockpit.dbus("org.freedesktop.NetworkManager", {});
+        this.proxy = this.client.proxy();
+    }
+
+    componentWillMount() {
+        const that = this;
+
+        this.proxy.wait(function() {
+            if (that.proxy.valid) {
+                
+                const devices = that.proxy.AllDevices;
+
+                for (let i in devices) {
+                    const deviceProxy = that.client.proxy("org.freedesktop.NetworkManager.Device", devices[i]);
```

Comment:
Should you maybe just proxy all the org.freedesktop.NetworkManager.Device instead. With client.proxies. That way you only have to add the listener once.

You should also close the proxies and the client when unmounted.